tracer.go

```diff
@@ -57,7 +57,7 @@ type Tracer interface {
 	// fails anyway.
 	//
 	// See Tracer.Join().
-	Inject(sp Span, format interface{}, carrier interface{}) error
+	Inject(sc SpanContext, format interface{}, carrier interface{}) error
```

Comment:
very important to note that Inject and Join are not symmetrical, type-wise... I think the API and semantics are much easier to reason about if `Join` returns a `Span` and `Inject` takes a `SpanContext`.